### ISSUE TITLE
Hot Module Replacement not working

### ISSUE THREAD

**Issue by rodp82** [aspnet/JavaScriptServices#603]
I've followed the tutorial at [http://blog.stevensanderson.com/2016/10/04/angular2-template-for-visual-studio/](url) and manged to get the application running, however, I don't see the "[HMR] connected" in the console. And when I try to make a change, the application doesn't update in the browser.

I've also tried setting up creating the application using the yo generator, but get the same issue.

Am I missing something else?
Thanks

**Comment by consigliory** [comment]
+1

**Comment by SteveSandersonMS** [comment]
That's strange - it certainly works as far as I'm aware, and I know lots of other people are using it with HMR.

Is it possible you're running in production mode, rather than development mode? (You need the environment variable `ASPNETCORE_ENVIRONMENT` set to `Development`, which is the case by default when you launch from VS, unless you've changed something). HMR is only turned on in development mode by default.

Do you have any repro steps that might enable me to see what's going on in your scenario? Of course, it would have to be something different than what's described in my blog post, since those steps lead to HMR working in all the cases I know of.

**Comment by consigliory** [comment]
I am using mac: 
npm install -g yo generator-aspnetcore-spa
yo aspnetcore-spa

**Comment by consigliory** [comment]
all is fine. You absolutely right. export ASPNETCORE_ENVIRONMENT=Development
now all is fine. Thanks a lot!

**Comment by MarkPieszak** [comment]
Were you using VSCode to launch? I believe the auto generated launch files maybe don't set it to DEV automatically.

But I believe you can add:
    "env": {
           "ASPNETCORE_ENVIRONMENT": "Development"
            },

To the configuration to ensure it.

**Comment by SteveSandersonMS** [comment]
Glad you got it figured out :)

**Comment by SteveSandersonMS** [comment]
Oh, just realised the original poster was @rodp82 not @consigliory.

I'll leave this closed because I'm pretty sure there isn't a bug here, but @consigliory, if you're still having trouble, please let us know.

**Comment by rodp82** [comment]
Thanks @SteveSandersonMS, I did have the ASPNETCORE_ENVIRONMENT set to development in windows using `set ASPNETCORE_ENVIRONMENT=development`
And in the browser console, it did have that message saying it was running in development mode. It just didn't output the `[HMR] connected` message after that as indicated in your tutorial.
I tried using Visual Studio 2015 Community Edition, Visual Studio Code and running it using dotnet run.

However, I just tried the same setup on my home PC and everything works as expected with no issues :(
Will keep poking around to see if I can figure it out.
